ShellPkg/Library/UefiShellAcpiViewCommandLib/Parsers/Spmi/SpmiParser.c

```diff
@@ -0,0 +1,94 @@
+/** @file
+  SPMI table parser
+
+  Copyright (C) 2024 Advanced Micro Devices, Inc. All rights reserved.
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+
+  @par Reference(s):
+    - IPMI 2.0 Specification - October 2013
+**/
+
+#include <IndustryStandard/Acpi.h>
+#include <Library/UefiLib.h>
+#include "AcpiParser.h"
+#include "AcpiTableParser.h"
+
+// Local variables
+STATIC ACPI_DESCRIPTION_HEADER_INFO  mAcpiHdrInfo;
+
+/**
+  Validate Interface Type.
+
+  @param [in] Ptr       Pointer to the start of the field data.
+  @param [in] Context   Pointer to context specific information e.g. this
+                        could be a pointer to the ACPI table header.
+**/
+STATIC
+VOID
+EFIAPI
+ValidateInterfaceType (
+  IN UINT8   *Ptr,
+  IN UINT32  Length,
+  IN VOID    *Context
+  )
+{
+  if (*Ptr > 5) {
+    IncrementErrorCount ();
+    Print (L"\nError: Invalid IPMI Interface type %d", *Ptr);
+  }
+}
+
+/// An ACPI_PARSER array describing the ACPI SPMI table.
+STATIC CONST ACPI_PARSER  SpmiParser[] = {
+  PARSE_ACPI_HEADER (&mAcpiHdrInfo),
+  { L"Interface Type",              1,   36, L"%d",   NULL,    NULL, ValidateInterfaceType, NULL },
+  { L"Reserved",                    1,   37, L"%x",   NULL,    NULL, NULL,                  NULL },
+  { L"Specification Revision",      2,   38, L"0x%x", NULL,    NULL, NULL,                  NULL },
+  { L"Interrupt Type",              1,   40, L"%d",   NULL,    NULL, NULL,                  NULL },

```

Comment:
I think you should check that:
`[7:2] - Reserved (must be 0)`

and to have a pretty print maybe, format the other bits with ParseAcpiBitFields():
```
[1] - I/O APIC/SAPIC interrupt (Global System Interrupt)
[0] - SCI triggered through GPE (use 0b for SSIF)
0 = not supported
1 = supported
```

Similar checks/formatting could be done regarding the GPE field:
```
 (Note: This field is valid only if Bit[0] of the InterruptType field is set.
```
and PCI Device Flag:
```
[7:1] - Reserved
[0] - PCI Device Flag. For PCI IPMI devices, this bit is set. For
non-PCI devices, this bit is cleared.
```

and other fields